plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/build/XWorkspaceBuilder.java

```diff
@@ -281,6 +288,27 @@ private BuildTask createBuildTask(WorkspaceChanges workspaceChanges) {
 		return deleted;
 	}
 
+	/**
+	 * Adds deltas to {@link #toBeConsideredDeltas} representing the deletion of all files in the deleted project to
+	 * ensure correct "isAffected"-computation on resource level during the next build (the deltas created here will be
+	 * considered as "external deltas" by {@link XStatefulIncrementalBuilder#launch()}).
+	 */
+	protected void handleDeletedProject(XIProjectConfig projectConfig) {
```

Comment:
True, it would be a bit more elegant in terms of code size. But the purpose explained in the comment includes the "mergeWithUnreportedDeltas"-call and it seems good to have everything related to that purpose in one place, imho.